### PR TITLE
Add overlaps for other reference region

### PR DESCRIPTION
**Pull request by arahuja**
[Review on Reviewable](https://reviewable.io/reviews/hammerlab/guacamole/279)

### CODE REVIEW

File: src/main/scala/org/hammerlab/guacamole/HasReferenceRegion.scala [hunk 1]
```diff
@@ -51,5 +51,14 @@ trait HasReferenceRegion {
...
     start - halfWindowSize <= locus && end + halfWindowSize > locus
   }
+
+  /**
+   * Does the region overlap another reference region
+   * @param other another region on the genome
+   * @return True if the the regions overlap
+   */
+  def overlaps(other: HasReferenceRegion): Boolean = {
+    other.referenceContig == referenceContig && (overlapsLocus(other.start) || overlapsLocus(other.end - 1))
```
Comment (timodonnell) [inline src/main/scala/org/hammerlab/guacamole/HasReferenceRegion.scala]: what if the other region completely encloses the current one? does this give the wrong result in that case?

e.g.
```
other: ------------XXXXXXXXXXXX----------
self:  -----------------XXXX--------------------

**Comment by timodonnell** [comment]
sorry for neglecting this for so long. if the issue I noted is indeed a bug, then I think we should add a test for that case in addition to fixing it. beside that LGTM

**Comment by arahuja** [comment]
Yea thanks - I had missed that, added a test and fixed the function

File: src/test/scala/org/hammerlab/guacamole/HasReferenceRegionSuite.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    val read2 = TestUtil.makeRead("TCGATCGA", cigar = "8M", mdtag = "8", start = 5L)
+
+    read1.overlaps(read2) should be(true)
+    read2.overlaps(read1) should be(true)
+
+  }
+
+  test("overlapping reads, different contigs") {
+    val read1 = TestUtil.makeRead("TCGATCGA", cigar = "8M", mdtag = "8", start = 1L)
+    val read2 = TestUtil.makeRead("TCGATCGA", cigar = "8M", mdtag = "8", start = 5L, chr = "chr2")
```
Comment (ryan-williams) [inline src/test/scala/org/hammerlab/guacamole/HasReferenceRegionSuite.scala]: there's an implied `{1-8,5-13} --> true` case here that it probably makes sense to add a case explicitly of

Comment (arahuja) [inline src/test/scala/org/hammerlab/guacamole/HasReferenceRegionSuite.scala]: @ryan-williams sorry for not looking at this sooner.  I don't think I understand what you mean - isn't the 1-8, 5-13 the test above where they are on the same chromosome?

Comment (ryan-williams) [inline src/test/scala/org/hammerlab/guacamole/HasReferenceRegionSuite.scala]: huh, yea.. I guess I misread / not sure what I was thinking. disregard

**Pull request merged**